include/decorators.h

```diff
@@ -55,6 +55,6 @@
   #if defined(__GNUC__) && __GNUC__ >= 7 || defined(__clang__) && __clang_major__ >= 12
```

Comment:
To remove ?
Maybe assert and refuse to compile if too old. I'm not sure the compilation would proceed successfully we've never tried and we don't want to support it

Reply:
As discussed, this will be addressed in another task / PR